java/dagger/internal/codegen/BindingGraph.java

```diff
@@ -1015,10 +1015,11 @@ void resolve(BindingKey bindingKey) {
         private final Set<Object> cycleChecker = new HashSet<>();
 
         /**
-         * Returns {@code true} if any of the bindings resolved for {@code bindingKey} are
-         * multibindings with contributions declared within this component's modules or optional
-         * bindings with present values declared within this component's modules, or if any of its
-         * unscoped dependencies depend on such bindings.
+         * Returns {@code true} if any of the bindings resolved for {@code bindingKey} are provision
+         * bindings that conflict with previously resolved bindings or multibindings with
```

Comment:
af890001 changes the comment on this method.

By checking the parent resolver to see if the binding was already resolved, we are, in effect comparing the subgraph to the one generated at the parent. Happy to explore this optimization in a later PR.